ORM/DataList.php

```diff
@@ -550,30 +535,48 @@ protected function isValidRelationName($field) {
 	}
 
 	/**
-	 * Translates a filter type to a SQL query.
+	 * Given an input string, parse it into the field name, filter type, and modifiers as
+	 * a trio.
 	 *
-	 * @param string $field - the fieldname in the db
-	 * @param string $filter - example StartsWith, relates to a filtercontext
-	 * @param array $modifiers - Modifiers to pass to the filter, ie not,nocase
-	 * @param string $value - the value that the filtercontext will use for matching
-	 * @return static
+	 * @param string $filter E.g. `Name:ExactMatch:not`, `Name:ExactMatch`, `Name:not`, `Name`
+	 * @return array List with the items field name, filter class, and list any of modifiers
 	 */
-	private function applyFilterContext($field, $filter, $modifiers, $value) {
-		if($filter) {
-			$className = "{$filter}Filter";
+	protected function parseFilterArguments($filter) {
+		// Field name is always the first component
+		$fieldArgs = explode(':', $filter);
+		$fieldName = array_shift($fieldArgs);
+
+		// Inspect type of second argument to determine context
+		$secondArg = array_shift($fieldArgs);
+		$modifiers = $fieldArgs;
+		if(!$secondArg) {
+			// No filter defaults to ExactMatchFilter
+			// E.g. `Name`
+			/** @skipUpgrade */
+			$filterClass = 'ExactMatchFilter';
+		} elseif(ucfirst($secondArg) === $secondArg) {
+			// Upper case pattern means filter class identifier
+			// E.g. `Name:ExactMatch` or `Name:ExactMatchFilter:not`
+			$filterClass = "{$secondArg}Filter";
```

Comment:
The second example here wouldn't work, right? It'd become `Name:ExactMatchFilterFilter:not`